### PR TITLE
Document moon orientation relative to sun

### LINKED ISSUE #11901: Moon texture is upside down
##### Minetest version
```
5.4.1
```

##### OS / Hardware

Operating system: Pop!_OS 21.10
CPU: i7 7700

GPU model: GTX 1060
OpenGL version: 4.6.0

##### Summary
Sun rising (about 6:00, ignore the size)
imagen

Moon rising (about 20:00, same texture)
imagen

##### Steps to reproduce
Apply the same texture both to sun and moon. Moon is upside down

### PR DESCRIPTION
**Pull request by appgurueu**
Closes #11901

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -6868,6 +6868,8 @@ object you are working with still exists.
             (default: `true`)
         * `texture`: A regular texture for the moon. Setting to `""`
             will re-enable the mesh moon. (default: `"moon.png"`)
+            Note: Relative to the sun, the moon texture is rotated by 180°.
```
Comment (rubenwardy) [inline doc/lua_api.txt:6871]:
```suggestion
            Note: Relative to the sun, the moon texture is rotated by 180°.
            This has the effect of the sun and the moon having the same orientation
            relative to their horizons when visible at the same time.
```

**Comment by Desour** [comment]
Could you please also add something like:
> * At time 5000, the moon texture appears non-rotated.

And something similar at the sun texture doc.

The note about the orientation of the moon relative to the sun can stay there as additional note.

**Comment by erlehmann** [comment]
> At time 5000, the moon texture appears non-rotated.

IMO the documentation (only?) needs to contain this:

```
At 05:00, both moon and sun appear right side up.
At 17:00, both moon and sun appear upside down.
```

I hope I got the times right.

**Comment by appgurueu** [comment]
Perhaps I should document the absolute rotation of moon & sun relative to the ground as they rise?

**Comment by Desour** [comment]
> Perhaps I should document the absolute rotation of moon & sun relative to the ground as they rise?

Yes, that's what I wanted to request. In my formulation I forgot take into account that potential future camera features could make "down" ambiguous.

**Comment by appgurueu** [comment]
> > Perhaps I should document the absolute rotation of moon & sun relative to the ground as they rise?
> 
> Yes, that's what I wanted to request. In my formulation I forgot take into account that potential future camera features could make "down" ambiguous.

I believe MT will always be Y-up.

**Comment by erlehmann** [comment]
> I believe MT will always be Y-up.

I disagree. Having modified minetest into having a center of gravity at (0,0,0) in the past, it seems straightforward to me to rotate the camera according to what the nearest center of gravity is.

**Review by rubenwardy — APPROVED** [review]

**Comment by appgurueu** [comment]
Rebased.

**Review by Df458 — APPROVED** [review]
Regardless of what else we may want to write on this topic what's here is already an improvement on what was already there. Since this is a documentation change, there's no harm in merging this and adding more later down the line if desired.

**Pull request merged**